services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/document/DefaultEnhancedDocument.java

```diff
@@ -60,12 +60,19 @@ public class DefaultEnhancedDocument implements EnhancedDocument {
 
     private final ChainConverterProvider attributeConverterProviders;
 
-    public DefaultEnhancedDocument(Map<String, AttributeValue> attributeValueMap,
+    private static final JsonItemAttributeConverter jsonConverter = JsonItemAttributeConverter.create();
+
+    private DefaultEnhancedDocument(Map<String, AttributeValue> attributeValueMap,
                                    ChainConverterProvider attributeConverterProviders) {
         this.attributeValueMap = attributeValueMap;
         this.attributeConverterProviders = attributeConverterProviders;
     }
 
+    public static DefaultEnhancedDocument fromAttributeValueMapAndConverters(Map<String, AttributeValue> attributeValueMap,
+                                                                             ChainConverterProvider attributeConverterProviders) {
+        return new DefaultEnhancedDocument(attributeValueMap, attributeConverterProviders);
```

Comment:
I was going to say that you can remove the extra ctor if you do
`return builder().attributeValueMap().attributeConverterProviders().build();` 
but I see that attributeValueMap isn't exposed on the builder. You could add attributeValueMap() as a private method on the builder if you prefer? Or keep as is.

Reply:
Good suggestion , since we will update the attributeValueMap internallu we can create a public setter only in DefaultEnhancedDocuement.Builder and not in EnhancedDocument.Builder